### PR TITLE
feature(core) adding normalizer Interface

### PR DESCRIPTION
**Pull request by wiewiurdp**
# Description

Adding normalizer Interface

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [ ] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/core/src/Application/Normalizer/JMSNormalizer.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+namespace Ergonode\Core\Application\Normalizer;
+
+use Ergonode\Core\Infrastructure\Exception\DenoralizationException;
+use Ergonode\Core\Infrastructure\Exception\NormalizerException;
+use JMS\Serializer\ArrayTransformerInterface;
+use JMS\Serializer\DeserializationContext;
+use JMS\Serializer\SerializationContext;
+use Psr\Log\LoggerInterface;
+
+class JMSNormalizer implements NormalizerInterface
```
Comment (piotrkreft) [inline module/core/src/Application/Normalizer/JMSNormalizer.php:19]: In my opinion, JMSSerializer should implement the interface and have the below implementation.

File: module/core/src/Infrastructure/Exception/DenoralizationException.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Core\Infrastructure\Exception;
```
Comment (piotrkreft) [inline module/core/src/Infrastructure/Exception/DenoralizationException.php:10]: Exceptions should be placed in application layer -> infrastructure is aware of application - not the other way around

File: module/core/src/Application/Normalizer/NormalizerInterface.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+declare(strict_types=1);
+
+namespace Ergonode\Core\Application\Normalizer;
+
+use JMS\Serializer\DeserializationContext;
+use JMS\Serializer\SerializationContext;
+
+interface NormalizerInterface
+{
+    public function normalize(object $data, ?SerializationContext $context, ?string $type): array;
```
Comment (piotrkreft) [inline module/core/src/Application/Normalizer/NormalizerInterface.php:17]: Remove the contexts please - you are binding the domain interface to the JMS implementation

**Review by piotrkreft — CHANGES_REQUESTED** [review]

File: module/core/src/Application/Normalizer/NormalizerInterface.php (hunk 3 above)
Comment (rprzedzik) [inline module/core/src/Application/Normalizer/NormalizerInterface.php:17]: You can't use JMS Context on ergonode Interface this connect it to JMS serizlaizer.

Comment (rprzedzik) [inline module/core/src/Application/Normalizer/NormalizerInterface.php:17]: Are this interface required type for ergonode normalize interface ??

**Review by rprzedzik — CHANGES_REQUESTED** [review]

File: module/core/src/Application/Normalizer/NormalizerInterface.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Core\Application\Normalizer;
```
Comment (piotrkreft) [inline module/core/src/Application/Normalizer/NormalizerInterface.php:10]: I think this could be a part of Normalizer bounded context

**Review by rprzedzik — APPROVED** [review]

**Review by piotrkreft — APPROVED** [review]

**Pull request merged**